api/controllers/game/ready.js

```diff
@@ -101,7 +103,9 @@ module.exports = function (req, res) {
           pNum: user.pNum,
           gameId: game.id,
         });
-        return Game.updateOne({ id: game.id }).set(gameUpdates);
+        const updateGamePromise = Game.updateOne({ id: game.id }).set(gameUpdates);
+        const unlockGamePromise = sails.helpers.unlockGame(lock);
+        return Promise.all([updateGamePromise, unlockGamePromise]);
```

Comment:
It occurred to me that it might make sense to eschew the `unlockGame()` call here and instead just add to the `gameUpdates` to set the lock to null. The only reason not to is so as not to overwrite someone elses' lock if ours has timed out since generally you only want a request to release its own lock (not someone else's). But it's a weird use case, so I'm not positive what's best.

Semantically, especially once we have async/await for this action, I like the simplicity of
```
const lock = await sails.helpers.lockGame(gameId);
// do stuff

await sails.helpers.unlockGame(lock);
```

though this does add an extra query

Reply:
I think it’s best to do the unlock and close the loop. We can investigate not needing one but it feels cleaner to actually remove it instead of unsetting it.